### PR TITLE
Add method `_generate_elicit_response_intent_and_annotations`

### LINKED ISSUE #135: Refactor generate_output in agenda_based_simulator
Add private methods to simplify the method `generate_output` (see inline TODOs).

### PR DESCRIPTION
**Pull request by NoB0**
Implementation `_generate_elicit_response_intent_and_annotations` to generate intent and annotations of the next simulator's utterance when the recommender system elicits preferences.

After retrieving the elicited slot and value, based on these, the `PreferenceModel` returns of preference or not that is used to generate the corresponding intent.

Follow-up issues:
- Support preference elicitation for multiple slot-value pairs (issue #141) 
- Handle pair with equal slot and value (DialogueKit issue [#234](https://github.com/iai-group/DialogueKit/issues/234))

Fixes #135

### CODE REVIEW

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 1]
```diff
@@ -59,6 +65,80 @@ def _generate_response(self, agent_utterance: Utterance) -> Utterance:
...
+            ) = self._preference_model.get_slot_preference(elicited_slot)
+
+            if response_value:
+                response_intent = self._interaction_model.INTENT_DISCLOSE  # type: ignore[attr-defined] # noqa
+                response_slot_values = [
+                    Annotation(slot=elicited_slot, value=response_value)
+                ]
+                return response_intent, response_slot_values
+
+        response_intent = self._interaction_model.INTENT_DONT_KNOW  # type: ignore[attr-defined] # noqa
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/agenda_based_simulator.py:139]: Nit: same as above

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 2]
```diff
@@ -85,13 +165,7 @@ def _generate_item_preference_response_intent(self, item_id: str) -> Intent:
...
         # should be based on the agenda).
         preference = self._preference_model.get_item_preference(item_id)
-        if preference > self._preference_model.PREFERENCE_THRESHOLD:
-            response_intent = self._interaction_model.INTENT_LIKE  # type: ignore[attr-defined] # noqa
-        elif preference < -self._preference_model.PREFERENCE_THRESHOLD:
-            response_intent = self._interaction_model.INTENT_DISLIKE  # type: ignore[attr-defined] # noqa
-        else:
-            response_intent = self._interaction_model.INTENT_NEUTRAL  # type: ignore[attr-defined] # noqa
-
+        response_intent = self._get_preference_intent(preference)
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/agenda_based_simulator.py:168]: Nit: return directly

**Review by kbalog — APPROVED** [review]
LGTM, nice work!

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py (hunk 2 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py]: Done.

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 3]
```diff
@@ -59,6 +65,80 @@ def _generate_response(self, agent_utterance: Utterance) -> Utterance:
...
+        else:
+            # Agent is asking about value preferences on a given slot, e.g.,
+            # "What movie genre would you prefer?"
+            (
+                response_value,
+                preference,
+            ) = self._preference_model.get_slot_preference(elicited_slot)
+
+            if response_value:
+                response_intent = self._interaction_model.INTENT_DISCLOSE  # type: ignore[attr-defined] # noqa
```
[the remark this answers is not in the document]
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py]: Done.

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py (hunk 1 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py]: Done.

**Pull request merged**